File: src/features/projects/types/project.ts
import { ProjectMetadata } from "./projectMetadata";

export interface Project {
  id: string;
  name: string;
  path: string;

  metadata: ProjectMetadata;

  favorite: boolean;

  createdAt: string;
  updatedAt: string;

  lastOpened?: string;

  gitBranch?: string;
  gitDirty: boolean;
}

File: src-tauri/src/models/project.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Serialize, Deserialize)]
pub struct ProjectDto {
    pub id: String,
    pub name: String,
    pub path: String,

    pub framework: String,
    pub language: String,

    pub favorite: bool,

    pub created_at: String,
    pub updated_at: String,
}